File: fp-java/src/main/java/cscie88a/week2/Dog.java
package cscie88a.week2;

public class Dog extends Animal implements ITrainable{

	public Dog() {
	}

	public Dog(String name, String eyeColor, String bodyColor) {
		super(name, eyeColor, bodyColor);
	}

	@Override
	public boolean doTrick(String trickName) {
		System.out.println(name + " says: I LOVE doing tricks! I'm doing "
				+ trickName + " now!");
		return true;
	}

}

File: fp-scala/src/main/scala/com/cscie88a/week2/Dog.scala
package com.cscie88a.week2

final case class Dog(
    name: String,
    eyeColor: String,
    bodyColor: String
  ) extends Animal(name, eyeColor, bodyColor)
       with Trainable {

  override def doTrick(trickName: String): Boolean = {
    println(
      s"${name} says - dogs love doing tricks. I'm doing a trick ${trickName} now"
    )
    true
  }

}
